gbpservice/nfp/orchestrator/modules/device_orchestrator.py

```diff
@@ -578,17 +591,10 @@ def _prepare_device_data_from_nfp_context(self, nfp_context):
         provider = nfp_context['provider']
         ports = []
 
-        if consumer['port']:
-            ports.append({
-                'id': consumer['port']['id'],
-                'port_classification': consumer['port_classification'],
-                'port_model': consumer['port_model']})
-
-        if provider['port']:
-            ports.append({
-                'id': provider['port']['id'],
-                'port_classification': provider['port_classification'],
-                'port_model': provider['port_model']})
+        if consumer['port_model'] == 'gbp_policy_target':
+            ports = self._make_ports_dict(consumer, provider, 'pt')
+        else:
+            ports = self._make_ports_dict(consumer, provider, 'port')
```

Comment:
Can we fill this info in node driver,
Its better to avoid such kind of checks in orchestrator.